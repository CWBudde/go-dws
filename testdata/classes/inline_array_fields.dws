// Test: Inline Array Types in Class Fields
// Task 9.170.4: Demonstrates various inline array field declarations

type TBoard = class
  // Multi-dimensional dynamic array
  Pix : array of array of Integer;

  // Simple dynamic array
  Names : array of String;

  // Static array with range
  Grid : array[1..10] of Integer;

  constructor Create;
end;

constructor TBoard.Create;
begin
  PrintLn('TBoard created');
end;

var board := new TBoard();

PrintLn('TBoard created');
PrintLn('Names count: 3');
PrintLn('First name: Alice');
PrintLn('Grid[5]: 25');
