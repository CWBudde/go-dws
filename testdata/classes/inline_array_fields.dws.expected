TBoard created
Names count: 3
First name: Alice
Grid[5]: 25
